src/utils/ReadOnlyException.ts

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { InternalException } from './InternalException';
+
+export class ReadOnlyException extends InternalException {
+    constructor(propertyName: string) {
+        super(`Attempting to set readonly property ${propertyName}`);
```

Comment:
I noticed you do this in the other PR and I don't know where you got this pattern of starting errors with a present tense "ing" verb. I generally see "Can't do this" or "This is not allowed" instead of "Trying to do this"

Reply:
I can't remember off the top of my head tbh, but I feel like I've seen that "ing" pattern before too. Anyway, I don't feel strongly about it, I renamed the error message to the one you had above